darwin/client.py

```diff
@@ -297,12 +73,19 @@ def list_remote_datasets(self, team: Optional[str] = None) -> Iterator[RemoteDat
         """
         Returns a list of all available datasets with the team currently authenticated against.
 
+        Parameters
+        ----------
+        team: Optional[str]
+            The team slug of the dataset. Defaults to None.
+
         Returns
         -------
-        list[RemoteDataset]
-        List of all remote datasets
+        Iterator[RemoteDataset]
+            List of all remote datasets
         """
-        for dataset in self.get("/datasets/", team=team):
+        response: List[Dict[str, Any]] = cast(List[Dict[str, Any]], self._get("/datasets/", team=team))
```

Comment:
Could you explain why we need `cast` here? Looks a bit too verbose for my taste

Reply:
I am making generalized use of cast because I want to tell Pylance that I know the type of variable. Basically, since we have `Union[A, B]` the only way to make Pylance accept that a given variable is of type `A` is to use a cast. 

I am open to alternatives if you know of any. This is what my research found out.

Reply:
Solved over Zoom talk.